engine/src/flutter/shell/platform/android/io/flutter/embedding/engine/systemchannels/SensitiveContentChannel.java

```diff
@@ -0,0 +1,141 @@
+              try {
+                sensitiveContentMethodHandler.isSupported(result);
+              } catch (IllegalStateException exception) {
+                result.error("error", exception.getMessage(), null);
+              }
+              break;
+            default:
+              Log.v(
+                  TAG, "Method " + method + " is not implemented for the SensitiveContentChannel.");
+              result.notImplemented();
+              break;
+          }
+        }
+      };
+
+  private int deserializeContentSensitivity(String contentSensitivityName) {
+    switch (contentSensitivityName) {
+      case AUTO_SENSITIVE_CONTENT_SENSITIVITY:
+        return View.CONTENT_SENSITIVITY_AUTO;
+      case SENSITIVE_CONTENT_SENSITIVITY:
+        return View.CONTENT_SENSITIVITY_NOT_SENSITIVE;
+      case NOT_SENSITIVE_CONTENT_SENSITIVITY:
+        return View.CONTENT_SENSITIVITY_SENSITIVE;
+      default:
+        throw new IllegalArgumentException(
+            "contentSensitivityName "
+                + contentSensitivityName
+                + " not known to the SensitiveContentChannel.");
+    }
+  }
+
+  private String serializeContentSensitivity(int contentSensitivityValue) {
+    switch (contentSensitivityValue) {
+      case View.CONTENT_SENSITIVITY_AUTO:
+        return AUTO_SENSITIVE_CONTENT_SENSITIVITY;
+      case View.CONTENT_SENSITIVITY_NOT_SENSITIVE:
+        return SENSITIVE_CONTENT_SENSITIVITY;
+      case View.CONTENT_SENSITIVITY_SENSITIVE:
+        return NOT_SENSITIVE_CONTENT_SENSITIVITY;
+      default:
+        throw new IllegalArgumentException(
+            "Android View content sensitivity constant with value "
+                + contentSensitivityValue
+                + " not known to the SensitiveContentChannel.");
+    }
+  }
+
+  public SensitiveContentChannel(@NonNull DartExecutor dartExecutor) {
+    channel =
+        new MethodChannel(dartExecutor, "flutter/sensitivecontent", StandardMethodCodec.INSTANCE);
+    channel.setMethodCallHandler(parsingMethodHandler);
+  }
+
+  /**
+   * Sets the {@link SensitiveContentMethodHandler} which receives all requests to get and set a
+   * particular content sensitivty level sent through this channel.
+   */
+  public void setSensitiveContentMethodHandler(
+      @Nullable SensitiveContentMethodHandler sensitiveContentMethodHandler) {
+    this.sensitiveContentMethodHandler = sensitiveContentMethodHandler;
+  }
+
+  public interface SensitiveContentMethodHandler {
+    /**
+     * Requests that a native Flutter Android {@code View} sets its content sensitivity level to
+     * {@code requestedContentSensitivity}.
+     */
+    void setContentSensitivity(
+        @NonNull int requestedContentSensitivity, @NonNull MethodChannel.Result result);
+
+    /** Returns the current content sensitivity level of a Flutter Android {@code View}. */
+    Integer getContentSensitivity();
+
+    /** Returns whether or not marking content sensitivity is supported on the device. */
```

Comment:
This method should also document if isSupported is static or dynamic. Should callers cache the result or can it change over time.